File: Digital Signal Processing. (MATLAB)/Labs/Lab 1/diffeqn.m
function [y] = diffeqn(x, a)
y = zeros(1, length(x));

for k = 1 : length(x)
    if k == 1
        y(1) = x(1);
    else
        y(k) = a*y(k-1) + x(k);
    end
end
end

File: Miscellaneous/Mathematics/diffeqn.py
"""
Digital Signal Processing Lab 1 diffeqn(x, a) function to solve difference equation (Matlab).
https://github.com/saudzahirr/BEE-19-SEECS/blob/main/Digital%20Signal%20Processing.%20(MATLAB)/Labs/Lab%201/diffeqn.m
"""

from numpy import *
from matplotlib.pyplot import *

def diffeq(x, a):
    y = []
    for k in range(len(x)):
        if k == 0:
            y.append(x[0])
        
        else:
            y.append(a*y[k-1] + x[k])
    
    print(y)
    stem(range(len(x)), y)
    show()

x = [2, 3, -1, 2, -3]
for i in zeros(5,):
    x.append(i)

diffeq(x, 2)




